### PR TITLE
controller-tools v0.2 rebase

### PR DESCRIPTION
**Pull request by damemi**
Updates to use v0.2.x of controller-tools, which fixes some bugs (such as https://bugzilla.redhat.com/show_bug.cgi?id=1712317), simplifies dependencies, and allows us to begin to unite our generator landscape under one tool, indivisible, with liberty and justice for all

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,7 +6,8 @@ RUN go build -o crd-schema-gen cmd/crd-schema-gen/main.go
 
 FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
 COPY --from=builder /go/src/github.com/openshift/crd-schema-gen/crd-schema-gen /
-COPY --from=builder /usr/local/go /usr/local/go
+#COPY --from=builder /usr/local/go /usr/local/go
+COPY --from=builder /usr /usr
```
Comment (damemi) [inline Dockerfile]: The problem I had was that the second stage is missing `gcc` (and some of its dependencies), which the parser requires. Without including gcc in the second stage I get 
```
go [list -e -json -compiled=true -test=false -export=false -deps=true -find=false -tags ignore_autogenerated -- ./582464517/pkg/...]: exit status 2: # runtime/cgo
exec: "gcc": executable file not found in $PATH
```
and just copying the `gcc` executable gives
```
go [list -e -json -compiled=true -test=false -export=false -deps=true -find=false -tags ignore_autogenerated -- ./053935065/pkg/...]: exit status 2: # runtime/cgo
gcc: error trying to exec 'cc1': execvp: No such file or directory
```


I've kept this as a separate commit to draw attention to it, hopefully there is a better solution.

**Comment by damemi** [comment]
This does seem to have some bugs still, eg optional and required tags might not be respected. Odd because raw controller-tools does respect them, but something in this generator does not

**Comment by damemi** [comment]
/hold

still some bugs with optional/nullable/required tags

**Comment by damemi** [comment]
see https://github.com/kubernetes-sigs/controller-tools/pull/228 for updates on things not currently working

**Comment by sttts** [comment]
/approve
/lgtm
/hold

Unhold when you think it's ok.

**Comment by damemi** [comment]
/hold cancel
bugs fixed

**Comment by damemi** [comment]
This is suddenly all broken again and I have no idea why this time. Looking into

**Comment by damemi** [comment]
Note this will require some updates to the tags currently in the API. Notably we'll need to add the `// +kubebuilder:validation:Optional` package-level default switcher to achieve parity with current generator expectations of a default to optional

**Comment by damemi** [comment]
This should be published to `registry.svc.ci.openshift.org/ocp/4.2:crd-schema-gen`

**Pull request merged**